credentials/tls/certprovider/provider.go

```diff
@@ -58,8 +58,9 @@ func getBuilder(name string) Builder {
 
 // Builder creates a Provider.
 type Builder interface {
-	// Build creates a new provider with the provided config.
-	Build(StableConfig) Provider
+	// Build creates a new Provider and initializes it with the given config and
+	// options combination.
+	Build(StableConfig, KeyMaterialOptions) Provider
```

Comment:
Should we name this `BuildOptions` or `ProviderOptions` instead of `KeyMaterialOptions` for extensibility?  (I slightly prefer `ProviderOptions`.  Since it's the `certprovider` package, `certprovider.Options` sounds plausible, too.)

Reply:
Renamed to `Options`.